runtime/nvqir/cudensitymat/CuDensityMatSim.cpp

```diff
@@ -80,16 +81,18 @@ class CuDensityMatSim : public nvqir::CircuitSimulatorBase<double> {
   CuDensityMatSim() {
     int numDevices{0};
     HANDLE_CUDA_ERROR(cudaGetDeviceCount(&numDevices));
-    const int deviceId =
-        cudaq::mpi::is_initialized() ? cudaq::mpi::rank() % numDevices : 0;
+    int currentDevice = -1;
```

Comment:
Is this magic number well-defined? Should really use an identifier so people can find it uses.